src/source/Signaling/LwsApiCalls.c

```diff
@@ -884,44 +971,47 @@ STATUS createChannelLws(PSignalingClient pSignalingClient, UINT64 time)
     PRequestInfo pRequestInfo = NULL;
     CHAR url[MAX_URI_CHAR_LEN + 1];
     CHAR paramsJson[MAX_JSON_PARAMETER_STRING_LEN];
-    CHAR tagsJson[2 * MAX_JSON_PARAMETER_STRING_LEN];
-    PCHAR pCurPtr, pTagsStart, pResponseStr;
-    UINT32 i, strLen, resultLen;
-    INT32 charsCopied;
+    PCHAR pResponseStr;
+    UINT32 i, resultLen;
     PLwsCallInfo pLwsCallInfo = NULL;
-    jsmn_parser parser;
-    jsmntok_t tokens[MAX_JSON_TOKEN_COUNT];
-    UINT32 tokenCount;
+    SignalingResult_t retSignal;
+    SignalingRequest_t signalRequest;
+    SignalingAwsRegion_t awsRegion;
+    CreateSignalingChannelRequestInfo_t createSignalingChannelRequestInfo;
+    SignalingChannelArn_t channelArn;
+    SignalingTag_t tags[MAX_TAG_COUNT];
 
     CHK(pSignalingClient != NULL, STATUS_NULL_ARG);
-
+    CHK(pSignalingClient->pChannelInfo->tagCount <= MAX_TAG_COUNT, STATUS_INVALID_ARG);
+
+    // Prepare AWS region
+    awsRegion.pAwsRegion = pSignalingClient->pChannelInfo->pRegion;
+    awsRegion.awsRegionLength = STRLEN(pSignalingClient->pChannelInfo->pRegion);
+    // Prepare URL buffer
+    signalRequest.pUrl = &url[0];
+    signalRequest.urlLength = MAX_URI_CHAR_LEN + 1;
+    // Prepare body buffer
+    signalRequest.pBody = &paramsJson[0];
+    signalRequest.bodyLength = MAX_JSON_PARAMETER_STRING_LEN;
     // Create the API url
-    STRCPY(url, pSignalingClient->pChannelInfo->pControlPlaneUrl);
-    STRCAT(url, CREATE_SIGNALING_CHANNEL_API_POSTFIX);
-
-    tagsJson[0] = '\0';
-    if (pSignalingClient->pChannelInfo->tagCount != 0) {
-        // Prepare the tags elements. Format the list at the end
-        pCurPtr = tagsJson + MAX_JSON_PARAMETER_STRING_LEN;
-        pTagsStart = pCurPtr;
-        for (i = 0; i < pSignalingClient->pChannelInfo->tagCount; i++) {
-            charsCopied = SNPRINTF(pCurPtr, MAX_JSON_PARAMETER_STRING_LEN - (pCurPtr - pTagsStart), TAG_PARAM_JSON_OBJ_TEMPLATE,
-                                   pSignalingClient->pChannelInfo->pTags[i].name, pSignalingClient->pChannelInfo->pTags[i].value);
-            CHK(charsCopied > 0 && charsCopied < MAX_JSON_PARAMETER_STRING_LEN - (pCurPtr - pTagsStart), STATUS_INTERNAL_ERROR);
-            pCurPtr += charsCopied;
-        }
-
-        // Remove the tailing comma
-        *(pCurPtr - 1) = '\0';
-
-        // Prepare the json params for the call
-        SNPRINTF(tagsJson, MAX_JSON_PARAMETER_STRING_LEN, TAGS_PARAM_JSON_TEMPLATE, tagsJson + MAX_JSON_PARAMETER_STRING_LEN);
-    }
-
-    // Prepare the json params for the call
-    SNPRINTF(paramsJson, ARRAY_SIZE(paramsJson), CREATE_CHANNEL_PARAM_JSON_TEMPLATE, pSignalingClient->pChannelInfo->pChannelName,
-             getStringFromChannelType(pSignalingClient->pChannelInfo->channelType),
-             pSignalingClient->pChannelInfo->messageTtl / HUNDREDS_OF_NANOS_IN_A_SECOND, tagsJson);
+    memset(&createSignalingChannelRequestInfo, 0, sizeof(CreateSignalingChannelRequestInfo_t));
```

Comment:
KVS SDK macro usage of MEMSET and SIZEOF

Reply:
Updated